samcli/commands/init/__init__.py

```diff
@@ -90,7 +90,7 @@
     help="Disable Cookiecutter prompting and accept default values defined template config",
 )
 @click.option(
-    "--extra_context",
+    "--extra-context",
```

Comment:
while this would be more consistent with the other SAM CLI parameters, it may be too late to change this since this was already released as v 0.31.0 :(

Reply:
Talked with @awood45 on this. Consistency with all our other options is important. This is a new field so we should adjust it now.